flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlToOperationConverterTest.java

```diff
@@ -1213,6 +1218,54 @@ public void testCreateTableWithMetadataColumn() {
         assertThat(actualSchema).isEqualTo(expectedSchema);
     }
 
+    @Test
+    public void testCreateFunction() {
+        // test create catalog function
+        String sql =
+                "CREATE FUNCTION test_udf AS 'org.apache.fink.function.function1' "
+                        + "LANGUAGE JAVA USING JAR 'file:///path/to/test.jar'";
+        final FlinkPlannerImpl planner = getPlannerBySqlDialect(SqlDialect.DEFAULT);
+        Operation operation = parse(sql, planner, getParserBySqlDialect(SqlDialect.DEFAULT));
+        assertThat(operation).isInstanceOf(CreateCatalogFunctionOperation.class);
+        CatalogFunction actualFunction =
+                ((CreateCatalogFunctionOperation) operation).getCatalogFunction();
+
+        assertThat(operation.asSummaryString())
+                .isEqualTo(
+                        "CREATE CATALOG FUNCTION: (catalogFunction: [Optional[This is a user-defined function]], "
+                                + "identifier: [`builtin`.`default`.`test_udf`], ignoreIfExists: [false], isTemporary: [false])");
+
+        // here doesn't assert the CatalogFunction directly because of the isGeneric method will
+        // load the class
+        assertThat(actualFunction.getClassName()).isEqualTo("org.apache.fink.function.function1");
```

Comment:
how about extract the assert logic to a common method so that the following assert can reuse it.

Reply:
+1. Can we add the equal function for CatalogFunctionImpl?

Reply:
I think it will be better if we provide this method.